galileo-egui/src/init.rs

```diff
@@ -28,7 +29,24 @@ impl MapApp {
 impl eframe::App for MapApp {
     fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {
         egui::CentralPanel::default().show(ctx, |ui| {
-            self.map.render(ui);
+            let attribution = self.map.collect_attributions();
+            if !attribution.text.is_empty() {
+                egui::Window::new("Attribution")
+                    .collapsible(false)
+                    .resizable(false)
+                    .title_bar(false)
+                    .anchor(Align2::RIGHT_BOTTOM, [-10., -10.])
+                    .fixed_size([350., 150.])
+                    .show(ui.ctx(), |ui| {
+                        ui.hyperlink_to(
+                            self.map.collect_attributions().text,
+                            self.map.collect_attributions().url.expect("URL error"),
+                        );
+                    });
+                self.map.render(ui);
+            } else {
+                self.map.render(ui);
+            }
```

Comment:
This logic must be moved to the `EguiMapState::render` function. Allso, you don't need to put everything into the if/state, just check if there are attribution, and add the attribution window in that case something along the lines:
```
pub fn render(&mut self, ui: &mut egui::Ui) {
    if let Some(attributions) = self.collect_attributions() {
        self.show_attributions(ui);
    }
    ....
}

fn show_attributions(&mut self, ui: &mut egui::Ui) {
    egui::Window::new("Attribution")
    ....
}
```